js/tests/components/SnippetPreviewSection.test.js

```diff
@@ -19,3 +19,65 @@ describe( "SnippetPreviewSection", () => {
 		expect( tree ).toMatchSnapshot();
 	} );
 } );
+
+describe( "mapEditorDataToPreview", () => {
+	it( "Hyphenates a space in the URL.", () => {
+		const exampleURL = "my URL";
+		const expected = "my-URL";
+
+		const dataObject = {
+			title: "",
+			url: exampleURL,
+			description: "",
+		};
+
+		const actual = mapEditorDataToPreview( dataObject );
+
+		expect( actual.url ).toEqual( expected );
+	} );
+
+	it( "Hyphenates all spaces in the URL.", () => {
+		const exampleURL = "my URL is awesome";
+		const expected = "my-URL-is-awesome";
+
+		const dataObject = {
+			title: "",
+			url: exampleURL,
+			description: "",
+		};
+
+		const actual = mapEditorDataToPreview( dataObject );
+
+		expect( actual.url ).toEqual( expected );
+	} );
+
+	it( "Doesn't hyphenate a trailing space.", () => {
```

Comment:
Same as above, though I could see this going wrong with if we accidentally hardcode to only remove hyphens at the last index, so perhaps these two tests are more useful than the one with spaces?